pkg/resources/istioingress/gateway.go

```diff
@@ -24,20 +24,30 @@ import (
 	"github.com/banzaicloud/kafka-operator/api/v1beta1"
 	"github.com/banzaicloud/kafka-operator/pkg/resources/templates"
 	"github.com/banzaicloud/kafka-operator/pkg/util"
-	kafkautil "github.com/banzaicloud/kafka-operator/pkg/util/kafka"
+	kafkautils "github.com/banzaicloud/kafka-operator/pkg/util/kafka"
 )
 
-func (r *Reconciler) gateway(log logr.Logger, externalListenerConfig v1beta1.ExternalListenerConfig) runtime.Object {
+func (r *Reconciler) gateway(log logr.Logger, externalListenerConfig v1beta1.ExternalListenerConfig,
+	_ v1beta1.IngressConfig, ingressConfigName, defaultIngressConfigName string) runtime.Object {
+	var gatewayName string
+	if ingressConfigName == util.IngressConfigGlobalName {
+		gatewayName = fmt.Sprintf(gatewayNameTemplate, r.KafkaCluster.Name, externalListenerConfig.Name)
+	} else {
+		gatewayName = fmt.Sprintf(gatewayNameTemplateWithScope, r.KafkaCluster.Name, externalListenerConfig.Name, ingressConfigName)
+	}
 	return &v1alpha3.Gateway{
-		ObjectMeta: templates.ObjectMeta(fmt.Sprintf(gatewayNameTemplate, r.KafkaCluster.Name, externalListenerConfig.Name), labelsForIstioIngress(r.KafkaCluster.Name, externalListenerConfig.Name), r.KafkaCluster),
+		ObjectMeta: templates.ObjectMeta(gatewayName,
+			labelsForIstioIngress(r.KafkaCluster.Name, annotationName), r.KafkaCluster),
```

Comment:
Why `annotationName` is passed into `labelsForIstioIngress` instead of `externalListenerConfig.Name`?